hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/functional/TestHoodieBackedMetadata.java

```diff
@@ -1871,7 +1865,11 @@ private void testTableOperationsImpl(HoodieSparkEngineContext engineContext, Hoo
       validateMetadata(client);
 
       // Restore
-      client.restoreToInstant("20210101000600000", writeConfig.isMetadataTableEnabled());
+      if (metaClient.getTableType() == COPY_ON_WRITE) {
+        assertThrows(HoodieRestoreException.class, () -> client.restoreToInstant("20210101000600000", writeConfig.isMetadataTableEnabled()));
+      } else {
+        client.restoreToInstant("20210101000600000", writeConfig.isMetadataTableEnabled());
+      }
```

Comment:
There should not be a need to check this based on table type. Need to look into why this fails for COW.